iac-extensions/arm/src/test/resources/checks/HardcodeApiVersionCheck/resources.json

```diff
@@ -0,0 +1,31 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+  "parameters": {
+    "customApiVersion": {
+      "type": "string"
+    }
+  },
+  "variables": {
+    "customApiVersion": "[first(providers(‘Microsoft.Compute’,’virtualMachines’).apiVersions)]"
+  },
+  "resources": [
+    {
+      "apiVersion": "[parameters('customApiVersion')]",
```

Comment:
I would add the test case where the apiVersion is `[first(providers(‘Microsoft.Compute’,’virtualMachines’).apiVersions)]`.
This should correctly raise an issue on your side.